### PR TITLE
improve packaging process and prevent `config.json` cooking for mobile

### PR DESCRIPTION
**Pull request by vaab**
Because of a small incompatibility in previous request method, iOS platform
could not download properly the `config.json`. As a consequence, for mobile
platform the `config.json` was baked in at build time. The standard `fetch()` API however
seem to be perfectly compatible, and is proposed so as to have a similar
deployment for web and mobile platforms and have a dynamic `config.json`
on all platform.

This also streamlines the process quite a bit and removes some
complexities.

Please note that the first commit comes from #211 as this PR is based on it.

### CODE REVIEW

**Review by seegwen — APPROVED** [review]
All in all I'm happy you did this changes, since at first I did not understand why you used this httpRequest method (did not know this lib, I'm more of a ky or directly fetch libs user) and as you mentioned in the RC, the sole purpose of this config.json mobile loading contorsion was because ios did not function properly with this lib.

**Comment by seegwen** [comment]
One more thought: I think it should be noted in the README that in order for changes in the config.json to take effect while assets have already been built, changes have to be done in `dist/config.json`.

File: README.md [hunk 1]
```diff
@@ -35,17 +35,26 @@ these instructions first:
...
+npm install
+```
+
+## Generating splash screens and icons
 
 Then put your icon and splash files in `resources/icon.png` and
 `resources/splash.png` respectively (see the files for minimum sizes)
 and then run:
 
 ```
```
[the remark this answers is not in the document]
Comment (vaab) [reply README.md:74]: Yep, make sense ! thanks ! I removed the "Then"

File: README.md [hunk 2]
```diff
@@ -35,17 +35,26 @@ these instructions first:
...
-cd project root
-cordova-res android --skip-config --copy && cordova-res ios --skip-config --copy
+npx cordova-res ${PLATFORM} --skip-config --copy
+```
+
+in which you'll have to replace `${PLATFORM}` with either `android` or
+`ios`.  Note that you can prepare both platforms by running this
+command twice, once for each platform.
+
 ```
```
[the remark this answers is not in the document]
Comment (vaab) [reply README.md:79]: I agree that this is not the best improvement... hmm, actually the argument is not needed, so I removed it and there are no more usage of `${PLATFORM}` anywhere.

**Comment by vaab** [comment]
I have added quite more information about the location of the `config.json` in the doc in a new commit. What is your thought about the new additions ? Thanks for your time !

**Review by seegwen — APPROVED** [review]
I think it's really good.

Thanks. Merging right away.

**Pull request merged**